numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,92 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    For some cases, there may be more efficient approaches
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create rolling window views.
+    shape : sequence of int
+        The shape of the new array.
+    step: sequence of int, optional
```

Comment:
I don't think this argument is useful - it's meaning is clearer when done via slicing after the fact

Reply:
that description is a mistake. It should be corrected as 
shape : sequence of int
        The shape of the sliding window.

Thanks for point out.